subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskDefinitionIntegrationSpec.groovy

```diff
@@ -260,170 +242,59 @@ class TaskDefinitionIntegrationSpec extends AbstractIntegrationSpec {
                 }
             }
 
-            task myTask(type: CustomTask, constructorArgs: ${constructorArgs})
+            tasks.create('myTask', CustomTask, 15)
         """
 
         when:
         run 'myTask'
 
         then:
         result.output.contains("got it 15")
-
-        where:
-        description | constructorArgs
-        'List'      | '[ 15 ]'
-        'Object[]'  | '[ 15 ] as Object[]'
     }
 
-    def "can construct a custom task without constructor arguments from plugin"() {
+    @Requires([KOTLIN_SCRIPT])
+    def 'can run custom task with constructor arguments via Project extension function'() {
         given:
-        file('buildSrc/src/main/java/CustomPlugin.java') << '''
-import org.gradle.api.*;
-import org.gradle.api.tasks.*;
-
-public class CustomPlugin implements Plugin<Project> {
-    public static class CustomTask extends DefaultTask {
-        @TaskAction
-        void printIt() {
-           System.out.println("hello world");
-        }
-    }
-
-    @Override
-    public void apply(Project p) {
-        p.getTasks().create("myTask", CustomTask.class);
-    }
-}
-'''
-
-        file('buildSrc/src/main/resources/META-INF/gradle-plugins/custom.properties') << 'implementation-class=CustomPlugin'
-
-        buildFile << "apply plugin: 'custom'"
-
-        when:
-        run 'myTask'
-
-        then:
-        result.output.contains('hello world')
-    }
-
-    def "can construct a custom task with constructor arguments via API from a plugin"() {
-        given:
-        file('buildSrc/src/main/java/CustomPlugin.java') << '''
-import org.gradle.api.*;
-import org.gradle.api.tasks.*;
-import javax.inject.Inject;
-
-public class CustomPlugin implements Plugin<Project> {
-    public static class CustomTask extends DefaultTask {
-        private final String message;
-        private final int number;
-
-        @Inject
-        public CustomTask(String message, int number) {
-            this.message = message;
-            this.number = number;
-        }
-
-        @TaskAction
-        void printIt() {
-            System.out.println(message + " " + number);
-        }
-    }
+        settingsFile << "rootProject.buildFileName = 'build.gradle.kts'"
+        file("build.gradle.kts") << """
+            import org.gradle.api.*
+            import org.gradle.api.tasks.*
+            import javax.inject.Inject
 
-    @Override
-    public void apply(Project p) {
-        p.getTasks().create("myTask", CustomTask.class, "hello", 42);
-    }
-}
-'''
+            inline
+            fun <reified T : Task> Project.task(name: String, vararg arguments: Any?) =
```

Comment:
Same as below. I think the method from Kotlin should be:
```
project.tasks.create<MyTask>("taskName", 15, "message")
```

Reply:
@eskatos @bamboo What do we need to do to make ☝️ work from Kotlin DSL?

Reply:
based on slack conversation, i've update to use `tasks.create<>`